writer.go

```diff
@@ -181,28 +234,70 @@ func (w *Writer) writeCheckDetailAddendum(cd *CheckDetail) error {
 
 // writeCheckImageView writes ImageViews (Detail, Data, Analysis) to a CheckDetail
 func (w *Writer) writeCheckImageView(cd *CheckDetail) error {
-	for _, ivDetail := range cd.GetImageViewDetail() {
-		if err := w.writeLine(ivDetail.String()); err != nil {
-			return err
-		}
+
+	ivDetailSlice := cd.GetImageViewDetail()
+	ivDataSlice := cd.GetImageViewData()
+	ivAnalysisSlice := cd.GetImageViewAnalysis()
+
+	// TODO: Add validator to ensure that each imageViewDetail has a corresponding imageViewData and imageViewAnalysis
+	// for now enforce that all images have data and analysis or no images have data and analysis
```

Comment:
@adamdecaf I don't see an obvious way of validating this without changing how ImageViewDetail/Data/Analysis are tracked.

Maybe something like this?
```
type CheckDetail struct {
...
imageViewDetail []ImageViewDetail
...
}

type ImageViewDetail struct {
...
imageViewData ImageViewData
imageViewAnalysis imageViewAnalysis
...
}
```

Reply:
It seems like CheckDetail needs to iterate through each Detail to find the matching Data and Analysis. Then we do a length check to make sure no missing/extra records are added.

We can do that check in another PR. I'd suggest a `ValidateOpts` struct with a field that allows us to disable such a check - I can see a vendor breaking that agreement.